rules/manager.go

```diff
@@ -423,12 +432,32 @@ func (g *Group) run(ctx context.Context) {
 					g.metrics.IterationsScheduled.WithLabelValues(GroupKey(g.file, g.name)).Add(float64(missed))
 				}
 				evalTimestamp = evalTimestamp.Add((missed + 1) * g.interval)
+
+				useAlertsActiveAtOverrideFunc(g)
+
 				iter()
 			}
 		}
 	}
 }
 
+func useAlertsActiveAtOverrideFunc(g *Group) {
+	if g.alertsActiveAtOverrideFunc != nil {
+		for _, rule := range g.Rules() {
```

Comment:
All this logic could be done inside an alertsActiveAtOverrideFunc, is it correct?

Reply:
will it be safer to have the state changing remains at the prometheus repo?

Reply:
> All this logic could be done inside an alertsActiveAtOverrideFunc, is it correct?

will it be safer to have the state changing remains at the prometheus repo?

Reply:
will it be safer to have the state changing remains at the prometheus repo?

Reply:
since there is no locking involved, it does not change much, but doing it like this might limit you in the future. If you prefer to have this here, it's fine for me.

Reply:
yeah we do prefer leave the override here to keep the override func to be more stateless

Reply:
@roidelapluie it's a nice call out about the future limitation. I'm curious about if you have examples that illustrates the limitations you are thinking?

Reply:
I think it's fine for me. I don't have examples right now.